src/ik_evolution_2.cpp

```diff
@@ -649,9 +651,18 @@ template <int memetic> struct IKEvolution2 : IKBase
     virtual size_t concurrency() const { return 4; }
 };
 
-static IKFactory::Class<IKEvolution2<0>> bio2("bio2");
-static IKFactory::Class<IKEvolution2<'q'>> bio2_memetic("bio2_memetic");
-static IKFactory::Class<IKEvolution2<'l'>> bio2_memetic_l("bio2_memetic_l");
+std::optional<std::unique_ptr<IKSolver>> makeEvolution2Solver(
+    const IKParams& params) {
+  const auto& name = params.solver_class_name;
+  if (name == "bio2")
+    return std::make_unique<IKEvolution2<0>>(params);
+  else if (name == "bio2_memetic")
```

Comment:
Because each `if` results in a `return` you don't need any `else` statements.

Reply:
Addressed in [e9c01ee](https://github.com/PickNikRobotics/bio_ik/pull/1/commits/e9c01ee4ef00ced5d79bb788ab01a067e93e922c)